### PR TITLE
Fix DescriptionField customization

### LINKED ISSUE #2481: Custom DescriptionField not working, Unable to render Icons with Description in Form
### Prerequisites

- [Yes] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [ ] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [ ] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

I am currently using the @rjsf/fluent-ui library to render Forms in the browser as part of an intern project. I created a custom DescriptionField to render Info icons with descriptions in my form but they do not seem to be working unless the property in question is of type: object. I am able to render icons with description if I simply render <DescriptionField description="foo" /> but not when I am trying to pass the DescriptionField into a Form using its fields prop. I've seen a couple other issues raised that are similar, and was wondering if there is a solution?

#### Expected behavior

Screenshot (36)

#### Actual behavior

Screenshot (37)

### Version

@rjsf/fluent-ui - ^2.5.1

### PR DESCRIPTION
**Pull request by alex-pex**
### Reasons for making this change

Using a description with a FixedArray and a custom theme (mui, boostrap, ...) crashes.
This fix the problem at the source. That's why it replaces https://github.com/rjsf-team/react-jsonschema-form/pull/2451

Fixes #2481, #2363, #2219

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by alex-pex** [comment]
Hi! I updated the PR, can someone do a review? I'm using a patched version for the moment

File: packages/core/src/components/fields/ArrayField.js [hunk 1]
```diff
@@ -717,6 +717,7 @@ class ArrayField extends Component {
     const arrayProps = {
       canAdd: this.canAddItem(items) && additionalSchema,
       className: "field field-array field-array-fixed-items",
+      DescriptionField,
```
Comment (epicfaace) [inline packages/core/src/components/fields/ArrayField.js:720]: Can you add a test please? Thanks!!

File: packages/bootstrap-4/test/Array.test.tsx [hunk 2]
```diff
@@ -47,4 +68,32 @@ describe("array fields", () => {
...
+        <div id="custom-title">{title}</div>
+      ),
+      DescriptionField: ({ description }: any) => (
+        <div id="custom-description">{description}</div>
+      ),
+    }
+    const tree = renderer
+      .create(<Form schema={schema} uiSchema={uiSchema} fields={fields} />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (epicfaace) [reply packages/bootstrap-4/test/Array.test.tsx:97]: I don't understand. Does it pass now? I don't see any CI failures.

File: packages/bootstrap-4/test/Array.test.tsx [hunk 3]
```diff
@@ -47,4 +68,32 @@ describe("array fields", () => {
       .toJSON();
     expect(tree).toMatchSnapshot();
   });
+  test("customization", () => {
```
Comment (epicfaace) [inline packages/bootstrap-4/test/Array.test.tsx:71]: make this test name more descriptive, so it's clear it's about description field customization with a fixed array

Comment (alex-pex) [reply packages/bootstrap-4/test/Array.test.tsx]: done!

**Comment by alex-pex** [comment]
@epicfaace do I need to do something else? for me the PR is ready for review (I updated the tests as asked)

**Comment by heath-freenome** [comment]
Fixed in the v5 beta, see the [5.x migration guide](https://react-jsonschema-form.readthedocs.io/en/latest/5.x%20upgrade%20guide/)

**Pull request closed without merging**